File: packages/types/src/project.ts
export interface Project {
  id: string
  name: string
  description?: string
  userId: string
  workspaceId: string
  status: 'active' | 'archived' | 'completed'
  tags: string[]
  skills: string[]
  gemIds: string[]
  connections: string[]
  chatIds: string[]
  documentIds: string[]
  memoryCount: number
  createdAt: string
  updatedAt: string
}

export interface Workspace {
  id: string
  name: string
  organizationId: string
  members: WorkspaceMember[]
  settings: WorkspaceSettings
  createdAt: string
  updatedAt: string
}

export interface WorkspaceMember {
  userId: string
  role: string
  joinedAt: string
}

export interface WorkspaceSettings {
  defaultModel?: string
  allowedModes: string[]
  maxProjects: number
  maxStorage: number
}

export interface Task {
  id: string
  projectId: string
  title: string
  description?: string
  status: 'todo' | 'in_progress' | 'done'
  priority: 'low' | 'medium' | 'high'
  assignedTo?: string
  createdAt: string
  updatedAt: string
}

File: services/api/app/schemas/project.py
from pydantic import BaseModel
from datetime import datetime


class ProjectCreate(BaseModel):
    name: str
    description: str | None = None
    tags: list[str] = []


class ProjectUpdate(BaseModel):
    name: str | None = None
    description: str | None = None
    status: str | None = None
    tags: list[str] | None = None


class ProjectResponse(BaseModel):
    id: str
    name: str
    description: str | None = None
    status: str
    tags: list[str] = []
    skills: list[str] = []
    chat_ids: list[str] = []
    document_ids: list[str] = []
    memory_count: int = 0
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True
